plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -90,17 +90,43 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 		if ( is_null( $element_max_intersection_ratio ) ) {
 			$processor->set_meta_attribute( 'unknown-tag', true ); // Mostly useful for debugging why an IMG isn't optimized.
 		} else {
-			// Otherwise, make sure visible elements omit the loading attribute, and hidden elements include loading=lazy.
+			// TODO: Take into account whether the element has the computed style of visibility:hidden, in such case it should also get fetchpriority=low.
 			$is_visible = $element_max_intersection_ratio > 0.0;
-			$loading    = $get_attribute_value( 'loading' );
-			if ( $is_visible && 'lazy' === $loading ) {
-				$processor->remove_attribute( 'loading' );
-			} elseif ( ! $is_visible && 'lazy' !== $loading ) {
+			if ( true === $context->url_metric_group_collection->is_element_positioned_in_any_initial_viewport( $xpath ) ) {
+				if ( ! $is_visible ) {
+					// If an element is positioned in the initial viewport and yet it is it not visible, it may be
+					// located in a subsequent carousel slide or inside a hidden navigation menu which could be
+					// displayed at any time. Therefore, it should get fetchpriority=low so that any images which are
+					// visible can be loaded with a higher priority.
+					$updated_fetchpriority = 'low';
+
+					// Also prevent the image from being lazy-loaded (or eager-loaded) since it may be revealed at any
+					// time without the browser having any signal (e.g. user scrolling toward it) to start downloading.
+					$processor->remove_attribute( 'loading' );
```

Comment:
Not sure this is needed? If I remember correctly, browser's lazy-loading implementations support horizontally offset images too, so shouldn't those still be lazy-loaded where applicable?

Worth noting that we may need to differentiate between "this is hidden by being offset outside the viewport" vs "this is simply hidden but technically would be in the viewport if shown".

Reply:
In this case, `fetchpriority=low` is what is recommended as opposed to `loading=lazy` as per [Optimize resource loading with the Fetch Priority API](https://web.dev/articles/fetch-priority#lower-carousel-priority):

> ![image](https://github.com/user-attachments/assets/0e935742-ec59-4493-a205-934d26660ed7)

And per [Optimize Largest Contentful Paint](https://web.dev/articles/optimize-lcp#:~:text=You%20can%20also%20lower%20the%20priority%20of%20images%20that%20might%20be%20early%20in%20the%20document%20response%20but%20aren%27t%20visible%20due%20to%20styling%2C%20such%20as%20images%20in%20carousel%20slides%20that%20aren%27t%20visible%20at%20startup%3A):

> ![image](https://github.com/user-attachments/assets/4cef623e-8f51-48c5-97f4-1c359a6fd167)

In the case of a carousel I believe you are right now that `loading=lazy` _may_ be OK, since the user can scroll as a signal to the browser that the image should start getting downloaded before displayed. However, the other case I have in mind are images which are hidden as being part of a mega menu, in which case they may be shown immediately in response to a user hovering over a nav menu item. In this scenario, `loading=lazy` would mean the image is not loaded when when the element is displayed. So I think `fetchpriority=low` is safer. Also, a carousel may not involve scrolling at all. The slides may be programmatically changed, so again there wouldn't be a user signal to start loading the image before the element is displayed.